web3-functions/gelato-cctp/index.ts

```diff
@@ -0,0 +1,153 @@
+import { ethers } from "ethers";
+import { join } from "./helper";
+import {
+  GelatoCCTPReceiver__factory,
+  GelatoCCTPSender__factory,
+  IMessageTransmitter__factory,
+} from "../../typechain";
+import {
+  Web3Function,
+  Web3FunctionContext,
+} from "@gelatonetwork/web3-functions-sdk";
+import {
+  ITransferWithAttestation,
+  AttestationStatus,
+  IRelayRequest,
+  IAttestation,
+  ITransfer,
+} from "./types";
+import {
+  CONSTANTS,
+  GELATO_API,
+  CIRCLE_API,
+} from "../../src/cctp-sdk/constants";
+import ky from "ky";
+
+(BigInt.prototype as any).toJSON = function () {
+  return this.toString();
+};
+
+Web3Function.onRun(async (context: Web3FunctionContext) => {
+  // index events from one network at a time
+  // execute transfers from one network at a time
+  const indexStr = await context.storage.get("index");
+  const index = indexStr ? parseInt(indexStr) : 0;
+
+  const nextindex = (index + 1) % Object.keys(CONSTANTS).length;
+  await context.storage.set("index", nextindex.toString());
+
+  const [chainId, constants] = Object.entries(CONSTANTS)[index];
+  const provider = await context.multiChainProvider.chainId(Number(chainId));
+
+  const toBlock = await provider.getBlockNumber();
+  const fromBlockStr = await context.storage.get(chainId);
+  const fromBlock = fromBlockStr ? parseInt(fromBlockStr) : toBlock;
```

Comment:
We need to make note that this event indexing logic has no sound REORG protection (e.g. via block confirmation delays). 

We should however at least make sure that we do exit the RUN if provider.getBlockNumber() == `fromBlock` from storage